ios/browser/api/sync/brave_sync_worker.cc

```diff
@@ -0,0 +1,430 @@
+    return true;
+  }
+
+  auto* local_device_info = GetLocalDeviceInfo();
+  if (!local_device_info) {
+    // May happens when we reset the chain immediately after connection
+    VLOG(1) << __func__ << " no local device info, cannot reset sync now";
+    return false;
+  }
+
+  auto* device_info_service =
+      DeviceInfoSyncServiceFactory::GetForBrowserState(browser_state_);
+  auto* tracker = device_info_service->GetDeviceInfoTracker();
+
+  if (!tracker) {
+    return false;
+  }
+    
+  tracker->DeleteDeviceInfo(local_device_info,
+                            base::BindOnce(&BraveSyncWorker::OnLocalDeviceInfoDeleted,
+                                           weak_ptr_factory_.GetWeakPtr()));
+
+  return true;
+}
+
+syncer::BraveProfileSyncService* BraveSyncWorker::GetSyncService() const {
+  DCHECK_CURRENTLY_ON(web::WebThread::UI);
+  return static_cast<syncer::BraveProfileSyncService*>(
+                 ProfileSyncServiceFactory::GetForBrowserState(browser_state_));
+}
+
+void BraveSyncWorker::OnStateChanged(syncer::SyncService* service) {
+  // If the sync engine has shutdown for some reason, just give up
+		if (!service || !service->IsEngineInitialized()) {
+				VLOG(3) << "[BraveSync] " << __func__ << " sync engine is not initialized";
+				return;
+		}
+  
+  SyncConfigInfo configuration = {};
+  if (!FillSyncConfigInfo(service, &configuration)) {
+    VLOG(3) << "[BraveSync] " << __func__
+            << " operations with passphrase are not required";
+    return;
+  }
+
+  brave_sync::Prefs brave_sync_prefs(browser_state_->GetPrefs());
+  std::string sync_code = brave_sync_prefs.GetSeed();
+  DCHECK_NE(sync_code.size(), 0u);
+  
+  if (!service->GetUserSettings()->IsEncryptEverythingAllowed()) {
+    configuration.encrypt_all = false;
+    configuration.set_new_passphrase = false;
+  }
+  
+  if (configuration.encrypt_all) {
+    service->GetUserSettings()->EnableEncryptEverything();
+  }
+  
+  bool passphrase_failed = false;
+  if (!sync_code.empty()) {
+    if (service->GetUserSettings()->IsPassphraseRequired()) {
+      passphrase_failed = !service->GetUserSettings()->SetDecryptionPassphrase(sync_code);
+    } else if (service->GetUserSettings()->IsTrustedVaultKeyRequired()) {
+      passphrase_failed = true;
+    } else {
+      if (configuration.set_new_passphrase &&
+          !service->GetUserSettings()->IsUsingSecondaryPassphrase()) {
+        service->GetUserSettings()->SetEncryptionPassphrase(sync_code);
+      }
+    }
+  }
+
+  if (passphrase_failed ||
+      service->GetUserSettings()->IsPassphraseRequiredForPreferredDataTypes()) {
+    VLOG(1) << __func__ << " setup passphrase failed";
+  }
+}
+
+void BraveSyncWorker::OnSyncShutdown(syncer::SyncService* service) {
+  if (sync_service_observer_.IsObserving(service)) {
+    sync_service_observer_.Remove(service);
+  }
+}
+
+void BraveSyncWorker::OnRemoteDeviceInfoDeleted() {
+  VLOG(0) << __func__ << " local device successfully removed";
```

Comment:
The log message is about the local device, but the function is about remote device